controller/src/beerocks/master/db/db.h

```diff
@@ -237,6 +237,15 @@ class db {
     void disconnected_slave_mac_queue_push(std::string node_mac);
     std::string node_to_string(std::string mac);
 
+    /**
+     * @brief Get the link metric data of a specific agent
+     *
+     * @param[in] mac The mac address of the db node that is used to save link metric data.
+     * @return reference to the map that holds link metric data of all agents.
+     */
+    std::unordered_map<sMacAddr, std::unordered_map<sMacAddr, son::node::link_metrics_data>> &
+    get_link_metric_data_map(const std::string &mac);
```

Comment:
Do you want to define another method with sMacAddr as an argument (with the same method name) or to replace this one completely?

Reply:
On second thought, as mentioned elsewhere, the argument is useless because it is always the GW node.

Instead, put this map in the db directly, not in the node, and no need for an argument in this function.